pkg/download/fetch.go

```diff
@@ -37,7 +37,7 @@ type HTTPFetcher struct{}
 func (HTTPFetcher) Get(uri string) (io.ReadCloser, error) {
 	resp, err := http.Get(uri)
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrapf(err, "failed to get the uri %q", uri)
```

Comment:
hmm do you think we could use wrapping `error` of FileFetcher.Get , like:

    failed to open specified archive file for reading: ..

Reply:
Makes total sense, then this is treated uniformly in both FileFetchers.